tools/library/CMakeLists.txt

```diff
@@ -35,6 +35,9 @@ include(GNUInstallDirs)
 set(CUTLASS_BUILD_MONO_LIBRARY OFF CACHE BOOL 
   "Determines whether the cutlass library is generated as a single file or multiple files.")
 
+option(BUILD_SHARED_LIBS "Build shared libraries" ON)
```

Comment:
`BUILD_SHARED_LIBS` is a global CMake flag and has side effects for many functions. Can you add the `CUTLASS_` prefix to these so we can limit the use to just CUTLASS code?